packages/selenium/test/RunPartial/AnalyzeTests.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using Deque.AxeCore.Commons;
+using Deque.AxeCore.Selenium;
+using NUnit.Framework;
+
+namespace Deque.AxeCore.Selenium.Test.RunPartial
+{
+    public class AnalyzeTests : TestBase
+    {
+        [Test]
+        public void ShouldReturnResults()
+        {
+            GoToFixture("index.html");
+
+            var res = new AxeBuilder(driver).Analyze();
+            Assert.IsNotNull(res);
+            Assert.IsNotNull(res.Inapplicable);
+            Assert.IsNotNull(res.Incomplete);
+            Assert.IsNotNull(res.Passes);
+            Assert.IsNotNull(res.Violations);
+        }
+
+        [Test]
+        public void ShouldThrowIfTopFrameErrors()
+        {
+            GoToFixture("crash.html");
+
+            var axe = new AxeBuilder(driver, CustomSource($"{axeSource}{Environment.NewLine}{axeCrashScript}"));
+            Assert.Throws<OpenQA.Selenium.JavaScriptException>(() => axe.Analyze());
+        }
+
+        [Test]
+        public void ShouldThrowWhenInjectingProblematicSource()
+        {
+            GoToFixture("crash.html");
```

Comment:
This should use a non-crashing fixture to isolate the behavior of crashing source vs crashing page, no?